### ISSUE TITLE
Running Vault on Docker for Mac

### ISSUE THREAD

**Issue by jicowan** [hashicorp/docker-vault#2]
I'm trying to run a dev instance of Vault on Docker for Mac. When I try running:
docker run -e 'VAULT_DEV_ROOT_TOKEN_ID=myroot' -e 'VAULT_DEV_LISTEN_ADDRESS=127.0.0.1:8200' vault
and get its status:
vault status
I get the following error.
Error checking seal status: Get http://127.0.0.1:8200/v1/sys/seal-status: dial tcp 127.0.0.1:8200: getsockopt: connection refused.

When I tried changing the listener address to 0.0.0.0, I got the same results.

**Comment by jefferai** [comment]
How are you running 'vault status'? You need to run it either in the container itself or via a mapped port. Given a container `angry_bell` started with your exact command:
```
$ docker exec angry_bell vault status -address="http://127.0.0.1:8200"
Sealed: false
Key Shares: 1
Key Threshold: 1
Unseal Progress: 0

High-Availability Enabled: false
```

**Comment by jefferai** [comment]
BTW, `/bin/sh` is available so you can also enter the container and run commands:
```
$ docker exec -it serene_goldwasser /bin/sh
/ # export VAULT_ADDR="http://127.0.0.1:8200"
/ # vault status
Sealed: false
Key Shares: 1
Key Threshold: 1
Unseal Progress: 0

High-Availability Enabled: false
```

**Comment by jicowan** [comment]
I have the vault client (CLI) running on my Mac.  I thought I'd be able to connect to the vault container that way.

**Comment by jefferai** [comment]
The Vault port is exposed in the container (https://github.com/hashicorp/docker-vault/blob/master/0.6/Dockerfile#L53) but it's up to the Docker operator to define the run-time mapping between an exposed port and the local port. Dockerfiles are not allowed to define/perform this action. See https://docs.docker.com/engine/userguide/networking/default_network/binding/ for more information.

**Comment by voanhduy1512** [comment]
Hi @jefferai 

I have the same issue when try to run Vault in Docker.

First, i started a vault container, exposed the port

```bash
docker run --name all_secret -p 8200:8200 vault
```
then check if it is running

```bash
docker exec deb74b978ff0 vault status -address="http://127.0.0.1:8200"

Sealed: false
Key Shares: 1
Key Threshold: 1
Unseal Progress: 0

High-Availability Enabled: false
```

```bash
docker ps
CONTAINER ID        IMAGE               COMMAND                  CREATED             STATUS              PORTS                    NAMES
123079cdadfa        vault               "docker-entrypoint.sh"   19 seconds ago      Up 18 seconds       0.0.0.0:8200->8200/tcp   all_secret
```
But when i access from my host machine
```bash
vault status -address="http://127.0.0.1:8200"
Error checking seal status: Get http://127.0.0.1:8200/v1/sys/seal-status: EOF
```

So even if i exposed the port to my host machine, vault still didn't accept my request. I try to find access log inside vault container but didn't find anything.

Do you have any ideas what is going wrong here?

**Comment by jefferai** [comment]
Not yet, but strangely, I can reproduce this on Ubuntu 16.04 too. I'm sure this worked previously. Will investigate.

**Comment by jefferai** [comment]
Blerg, somewhere along the way things changed and this broke. I'll fix it.

**Comment by jicowan** [comment]
I eventually got this to work by running: docker run -p 8200:8200 -e 'VAULT_DEV_ROOT_TOKEN_ID=<root_id>' -e 'VAULT_DEV_LISTEN_ADDRESS=0.0.0.0:8200' vault

**Comment by jefferai** [comment]
You could also bind to the host net. But by default it should listen on 0.0.0.0 in the container. I'm pretty sure I had logic for that in the entry script at some point and then lost it.

**Comment by jefferai** [comment]
Reopening this because although this fix worked for 0.6.0 it broke something for 0.6.1. Ref hashicorp/vault#1762. I have a fix but can't test it right now as Alpine's package servers are `503`ing. Argh...

**Comment by jefferai** [comment]
https://github.com/docker-library/official-images/pull/2086
